bson/bsonrw/value_writer.go

```diff
@@ -38,7 +38,7 @@ func NewBSONValueWriterPool() *BSONValueWriterPool {
 	return &BSONValueWriterPool{
 		pool: sync.Pool{
 			New: func() interface{} {
-				return new(valueWriter)
+				return newValueWriter(nil)
```

Comment:
Why did this not use `newValueWriter` before? /Were we not doing the appropriate setup before?

Reply:
This change is actually an artifact of fixing failing tests before I realized that `vw.reset()` needed to be called after each use of a `valueWriter` from the pool. It's not necessary or an improvement, so I can change it back.

Reply:
Reverted to simplify the change to only what's required.